docs/guides/server/db.adoc

```diff
@@ -128,20 +128,27 @@ Then continue configuring the database as described in the next section.
 
 For each supported database, the server provides some opinionated defaults to simplify database configuration. You complete the configuration by providing some key settings such as the database host and credentials.
 
-. Start the server and set the basic options to configure a database
+The configuration can be set during a *build command* OR a *start command*:
+
+. Using a *build command* followed by an optimzed start command (recommended)
++
+----
+bin/kc.[sh|bat] build --db postgres --db-url-host mypostgres --db-username myuser --db-password change_me
+bin/kc.[sh|bat] start --optimized
```

Comment:
Only the build time options can be specified on the build command

```suggestion
bin/kc.[sh|bat] build --db postgres 
bin/kc.[sh|bat] start --optimized --db-url-host mypostgres --db-username myuser --db-password change_me
```

Reply:
Ok, I missed that detail. The whole first point is to configure the database using only the build command. I would suggest to use the conf/keycloak.conf file instead to avoid having settings spread across 2 different commands.

Reply:
> Ok, I missed that detail. The whole first point is to configure the database using only the build command.

Unfortunately we're dealing with both build time and run time options. You configure build time database type with the build command, then the database related runtime options on the start command.

> I would suggest to use the conf/keycloak.conf file instead to avoid having settings spread across 2 different commands.

Right, most will use that or environment variables when creating Docker images - calling that out in general (not specifically for the optimized approach) will also address the concern about the password being shown directly on the cli.